povertymapping/rollout_grids.py

```diff
@@ -1,93 +1,292 @@
-import sys
 from povertymapping.geoboundaries import get_geoboundaries
 from povertymapping.hdx import get_hdx_file
 from geowrangler.grids import BingTileGridGenerator
 import geowrangler.spatialjoin_highest_intersection as sjhi
 import geowrangler.raster_zonal_stats as rzs
 import geopandas as gpd
+import pandas as pd
 import numpy as np
 from pathlib import Path
 import os
 from loguru import logger
+import fastcore.all as fc
+import rasterio as rio
+from tqdm import tqdm
 
 
-DEFAULT_ADMIN_LVL = 'ADM2'
+DEFAULT_ADMIN_LVL = "ADM2"
 DEFAULT_QUADKEY_LVL = 14
-DEFAULT_CACHE_DIR = '~/.cache/geowrangler'
-
-def get_region_filtered_bingtile_grids(region: str,
-                              admin_lvl = DEFAULT_ADMIN_LVL ,
-                              quadkey_lvl = DEFAULT_QUADKEY_LVL,
-                              use_cache=True,
-                              cache_dir=DEFAULT_CACHE_DIR,
-                              filter_population=True,
-                              assign_grid_admin_area=True,
-                              metric_crs='epsg:3857',
-                              extra_args=dict(
-                                nodata=np.nan),
+DEFAULT_CACHE_DIR = "~/.cache/geowrangler"
+
+
+def parallel_zonal_stats(batch_item):
+    batch, hdx_pop_file, aggregation, extra_args = batch_item
+    result = rzs.create_raster_zonal_stats(
+        batch, hdx_pop_file, aggregation=aggregation, extra_args=extra_args
+    )
+    return result
+
+
+def compute_parallel_raster_zonal_stats(
+    batch, hdx_pop_file, aggregation, extra_args, n_workers
+):
+    batch_items = [
+        (item.copy().reset_index(drop=True), hdx_pop_file, aggregation, extra_args)
+        for item in np.array_split(batch, n_workers)
+    ]
+    results = fc.parallel(
+        parallel_zonal_stats,
+        batch_items,
+        n_workers=n_workers,
+        threadpool=True,
+        progress=True,
+    )
+    logger.info(f"Completed parallel raster zonal stats for {len(results)} threads")
+    result = pd.concat(results, ignore_index=True)
+    logger.info(f"Concatenated parallel raster zonal stats for {len(results)} threads")
+    result = gpd.GeoDataFrame(result, geometry="geometry", crs=batch.crs)
+    return result
+
+
+def compute_raster_stats(
+    admin_grids_gdf,
+    hdx_pop_file,
+    aggregation=dict(column="population", output="pop_count", func="sum"),
+    extra_args=dict(nodata=np.nan),
+    group_col=None,
+    max_batch_size=None,
+    n_workers=None,
+):
+    "Compute raster stats"
```

Comment:
This function's pretty long. Nice to have a short summary about what this does (I think the key thing about this is it's an optimized way of computing raster stats right? Capable of handling big rasters and parallelizing for faster compute time).